pkg/config/keys.go

```diff
@@ -21,9 +22,15 @@ func MakeZoneKeyPrefix(id SystemTenantObjectID) roachpb.Key {
 	return keys.SystemSQLCodec.ZoneKeyPrefix(uint32(id))
 }
 
-// MakeZoneKey returns the key for id's entry in the system.zones table.
-func MakeZoneKey(id SystemTenantObjectID) roachpb.Key {
-	return keys.SystemSQLCodec.ZoneKey(uint32(id))
+// MakeZoneKey returns the key for a given id's entry in the system.zones table.
+func MakeZoneKey(codec keys.SQLCodec, id descpb.ID) roachpb.Key {
+	return codec.ZoneKey(uint32(id))
+}
+
+// MakeZoneKeyForSystemTenant returns the key for the given system tenant ID's
+// entry in the system.zones table.
+func MakeZoneKeyForSystemTenant(id SystemTenantObjectID) roachpb.Key {
+	return MakeZoneKey(keys.SystemSQLCodec, descpb.ID(id))
 }
 
 // DecodeSystemTenantObjectID decodes the object ID for the system-tenant from
```

Comment:
This function should also thread in a codec. Does it have to return this SystemTenantObjectID type?

Reply:
Better yet, can we get rid of the SystemTenantObjectID type entirely? Instead we could ensure we're using a codec + descpb.ID everywhere, specifying the SystemSQLCodec explicitly whenever needed.